### ISSUE TITLE
Basic NextJS starter : Error: Expected never to be called, but received: 320

### ISSUE THREAD

**Issue by Thisisjuke** [Thinkmill/keystatic#255]
I wanted to try this repository: 

![image](https://user-images.githubusercontent.com/23522065/234376985-b68a33f1-d31a-44e7-b3b9-5dd19ef8086f.png)

When I clone the Starter called "Start from a blank Canva", I can run the project locally but can't go to `/keystatic`.

When I click on the button [Go to Keystatic](http://localhost:3000/keystatic), I have the following error:

```
Unhandled Runtime Error
Error: Expected never to be called, but received: 320

Call Stack
assertNever
node_modules/emery/assertions/dist/emery-assertions.esm.js (25:0)
sizeResolver
node_modules/@voussoir/style/dist/voussoir-style.esm.js (689:15)
mapResponsiveValue
node_modules/@voussoir/style/dist/voussoir-style.esm.js (517:0)
convertStyleProps
node_modules/@voussoir/style/dist/voussoir-style.esm.js (912:0)
useStyleProps
node_modules/@voussoir/style/dist/voussoir-style.esm.js (941:0)
useFlexStyleProps
node_modules/@voussoir/layout/dist/voussoir-layout.esm.js (129:22)
eval
node_modules/@voussoir/layout/dist/voussoir-layout.esm.js (253:0)
renderWithHooks
node_modules/react-dom/cjs/react-dom.development.js (16305:0)
updateForwardRef
node_modules/react-dom/cjs/react-dom.development.js (19226:0)
beginWork
node_modules/react-dom/cjs/react-dom.development.js (21636:0)
HTMLUnknownElement.callCallback
node_modules/react-dom/cjs/react-dom.development.js (4164:0)
Object.invokeGuardedCallbackDev
node_modules/react-dom/cjs/react-dom.development.js (4213:0)
invokeGuardedCallback
node_modules/react-dom/cjs/react-dom.development.js (4277:0)
beginWork$1
node_modules/react-dom/cjs/react-dom.development.js (27451:0)
performUnitOfWork
node_modules/react-dom/cjs/react-dom.development.js (26557:0)
workLoopSync
node_modules/react-dom/cjs/react-dom.development.js (26466:0)
renderRootSync
node_modules/react-dom/cjs/react-dom.development.js (26434:0)
performConcurrentWorkOnRoot
node_modules/react-dom/cjs/react-dom.development.js (25738:0)
workLoop
node_modules/scheduler/cjs/scheduler.development.js (266:0)
flushWork
node_modules/scheduler/cjs/scheduler.development.js (239:0)
MessagePort.performWorkUntilDeadline
node_modules/scheduler/cjs/scheduler.development.js (533:0)
```

I set in package.json set NextJS to exact versions:
- next@13.1.1
- next@13.2.3

![image](https://user-images.githubusercontent.com/23522065/234378386-75bb2d12-d817-40cd-8047-88bf60802d68.png)

It's also the case in my personnal NextJS application (I tried the starter to see if it was relative to one of my middleware but It's not : the bug is present in the starter repo).

## Expected Behavior

I want to be able to go to `/keystatic` and have the dashboard / editor UI.

## Current Behavior

I have an error on the editor page.

## Steps to Reproduce

1. Go to: https://keystatic.thinkmill.com.au/
2. Chose the starter "Start from a blank canvas" and click on "get started".
3. Clone it to your personnal github (private) using Vercel interface.
4. Pull the repo locally.
5. Run `npm i` and `npm run dev`
6. Go to your starting app url and then click on "Go to keystatic".
7. You should see the `Error: Expected never to be called, but received: 320` error.

**Comment by JPrisk** [comment]
Hey @Thisisjuke,

I've added some additional information to the readme which will help you getting up and running. Without diving into the details of the `Unhandled Runtime Error` I think you are seeing this due to incompatible dependencies.

Are you able to try the steps you've outlined above using `pnpm`?

**Comment by Thisisjuke** [comment]
Hey JPrisk,

I tested the NextJS repository using `pnpm` and it's working fine.
I also tested it with `yarn` and I have the same error : `Expected never to be called, but received: 320`.

So I have another question: our app is dockerized and deployed with an orchestrator. I don't think we will be able to change all the deployment / CI-CD of the current app `npm` -> `pnpm` in a short term.

So at the moment, is it not possible to choose between `npm`, `yarn`, and `pnpm`? For example, we have another project using Yarn Workspace, so we just can't use pnpm, so keystatic is out of the question?

Additional information : 
- There was an issue in `next@13.3.1` relative to a sub-dependency of keystatic: https://github.com/vercel/next.js/issues/48593. That has been fixed in `next@13.3.2`.
- I tested keystatic with `pnpm`, `next@13.3.2` and just released `next@13.3.3` and I see no issues: It seems that I can see the dashboard and can use the dashboard to create, edit and delete.

**Comment by Thisisjuke** [comment]
Hey comming back to you after trying things own my side

I think keystatic is now working in my project environment. Perhaps, I still have issues with NextJS concerning the issue #48593 above. I can close the React Error Modal and play with keystatic normally, but it's kinda strange since I don't have this issue in the starter repository. 

I will continue to set up keystatic on my development environment as if this error would be quickly resolved on NextJS side. I will let you know if I have issues that seems relative to keystatic or keystatic+nextjs (not NextJS only)
